fuel-vm/src/interpreter/blob.rs

```diff
@@ -0,0 +1,91 @@
+use fuel_asm::{
+    RegisterId,
+    Word,
+};
+use fuel_storage::StorageSize;
+use fuel_tx::PanicReason;
+use fuel_types::BlobId;
+
+use crate::{
+    error::IoResult,
+    prelude::*,
+    storage::{
+        BlobData,
+        InterpreterStorage,
+    },
+};
+
+use super::{
+    internal::inc_pc,
+    split_registers,
+    GetRegMut,
+    Interpreter,
+    Memory,
+    SystemRegisters,
+    WriteRegKey,
+};
+
+impl<M, S, Tx, Ecal> Interpreter<M, S, Tx, Ecal>
+where
+    M: Memory,
+    S: InterpreterStorage,
+    <S as InterpreterStorage>::DataError: From<S::DataError>,
+{
+    pub(crate) fn blob_size(
+        &mut self,
+        dst: RegisterId,
+        blob_id_ptr: Word,
+    ) -> IoResult<(), S::DataError> {
+        self.gas_charge(self.interpreter_params.gas_costs.bsiz().base())?;
+
+        let blob_id = BlobId::from(self.memory.as_ref().read_bytes(blob_id_ptr)?);
+
+        let Some(size) =
+            <S as StorageSize<BlobData>>::size_of_value(&self.storage, &blob_id)
+                .map_err(RuntimeError::Storage)?
+        else {
+            return Err(PanicReason::BlobNotFound.into());
+        };
+        self.dependent_gas_charge_without_base(
+            self.interpreter_params.gas_costs.bsiz(),
+            size as Word,
+        )?;
+        let (SystemRegisters { pc, .. }, mut w) = split_registers(&mut self.registers);
+        let result = &mut w[WriteRegKey::try_from(dst)?];
+        *result = size as Word;
+        Ok(inc_pc(pc)?)
+    }
+
+    pub(crate) fn blob_load_data(
+        &mut self,
+        dst_ptr: Word,
+        blob_id_ptr: Word,
+        blob_offset: Word,
+        len: Word,
+    ) -> IoResult<(), S::DataError> {
+        self.dependent_gas_charge(self.interpreter_params.gas_costs.bldd(), len)?;
```

Comment:
Should we charge for the size of the contract as well, and not just the len copied? Essentially same question applies for the `bsiz`, which could possibly be constant cost.

Reply:
TODO: review code to check if we actualy do this correctly, i.e. it really should depend on the size